EIPS/eip-5727.md

```diff
@@ -0,0 +1,596 @@
+
+Our interface can be used to represent non-transferable ownerships, and provides features for common use cases including but not limited to:
+
+- granular lifecycle management of SBTs (e.g. minting, revocation, expiration)
+- management of SBTs via community voting and delegation (e.g. DAO governance, operators)
+- recovery of SBTs (e.g. switching to a new wallet)
+- token visibility control (e.g. private SBTs, hiding negative tokens)
+- fungible and non-fungible SBTs (e.g. membership card and loyalty points)
+- the grouping of SBTs using slots (e.g. complex reward schemes with a combination of vouchers, points, and badges)
+
+A common interface for soulbound tokens will not only help enrich the Web3 ecosystem but also facilitates the growth of a decentralized society.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Core
+```solidity
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+/**
+ * @title ERC5727 Soulbound Token Interface
+ * @dev The core interface. It allows basic query of information about tokens and slots.
+ */
+interface IERC5727 is IERC165 {
+    /**
+     * @dev MUST emit when a token is minted.
+     * @param soul The address that the token is minted to
+     * @param tokenId The token minted
+     * @param value The value of the token minted
+     */
+    event Minted(address indexed soul, uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is revoked.
+     * @param soul The owner soul of the revoked token
+     * @param tokenId The revoked token
+     */
+    event Revoked(address indexed soul, uint256 indexed tokenId);
+
+    /**
+     * @dev MUST emit when a token is charged.
+     * @param tokenId The token to charge
+     * @param value The value to charge
+     */
+    event Charged(uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is consumed.
+     * @param tokenId The token to consume
+     * @param value The value to consume
+     */
+    event Consumed(uint256 indexed tokenId, uint256 value);
+
+    /**
+     * @dev MUST emit when a token is destroyed.
+     * @param soul The owner soul of the destroyed token
+     * @param tokenId The token to destroy.
+     */
+    event Destroyed(address indexed soul, uint256 indexed tokenId);
+
+    /**
+     * @dev MUST emit when the slot of a token is set or changed.
+     * @param tokenId The token of which slot is set or changed
+     * @param oldSlot The previous slot of the token
+     * @param newSlot The updated slot of the token
+     */
+    event SlotChanged(
```

Comment:
I think your specification section is a little light on details. It might be useful to describe each interface briefly with some English text before diving into the solidity, and maybe explain a bit about the unique concepts in this EIP (like slots.)